### PR TITLE
[lint] Unused var and import error

### PR DESCRIPTION
**Pull request by cidrblock**
Fix an unused variable in actions/config
Use the same approach in 2 other places. (improves the fix in #746)

Disable the import error in utils, since that directory isn't really a package, but instead catalog_collections is just a script that gets run, so a full or relative import couldn't be used in this case.

### CODE REVIEW

**Review by ssbarnea — APPROVED** [review]

**Comment by cidrblock** [comment]
recheck

**Comment by ssbarnea** [comment]
recheck

File: share/ansible_navigator/utils/catalog_collections.py [hunk 1]
```diff
@@ -29,8 +29,10 @@
 except ImportError:
     from yaml import SafeLoader  # type: ignore
 
+# pylint: disable=import-error
 from key_value_store import KeyValueStore  # type: ignore
 
+# pylint: enable=import-error
```
Comment (webknjaz) [inline share/ansible_navigator/utils/catalog_collections.py:35]: Try to always disable things inline
```suggestion
from key_value_store import KeyValueStore  # type: ignore  # pylint: disable=import-error

```

Comment (cidrblock) [reply share/ansible_navigator/utils/catalog_collections.py:35]: cool, ty

File: src/ansible_navigator/actions/config.py [hunk 2]
```diff
@@ -321,7 +321,7 @@ def _parse_and_merge(self, list_output, dump_output) -> None:
                         parsed[variable]["via"] = source
                     parsed[variable]["current"] = current
                     parsed[variable]["__current_value"] = extracted.groupdict()["current"]
-                except KeyError as exc:  # noqa: F841
+                except KeyError as _exc:  # noqa: F841
                     self._logger.error("variable '%s' not found in list output")
```
Comment (webknjaz) [inline src/ansible_navigator/actions/config.py:325]: have you considered just adding `exc_info=exc` or dropping the `as` part?

File: share/ansible_navigator/utils/catalog_collections.py (hunk 1 above)
Comment (webknjaz) [reply share/ansible_navigator/utils/catalog_collections.py]: A bit more context: comments (any comments) tend to float around the code base. If you add a comment on top of a line, there's no guarantee that in the future, an unrelated line won't appear between a comment and its context. Putting comments on the same line has a better chance of surviving such situation and you won't see cases of misleading comments explaining something that doesn't seem related to the code below (if one line can sneak in, a block of code can too, just give it time).

This is why my rule of thumb is to move the pylint comment above the line only if there are technical limitations that prevent inlining, like linters complaining about the line being too long.

File: src/ansible_navigator/actions/config.py (hunk 2 above)
Comment (cidrblock) [reply src/ansible_navigator/actions/config.py:325]: ha! I didn't even think of that, I'll do another PR and look for others too.

File: src/ansible_navigator/actions/config.py [hunk 3]
```diff
@@ -321,7 +321,7 @@ def _parse_and_merge(self, list_output, dump_output) -> None:
                         parsed[variable]["via"] = source
                     parsed[variable]["current"] = current
                     parsed[variable]["__current_value"] = extracted.groupdict()["current"]
-                except KeyError as exc:  # noqa: F841
+                except KeyError as _exc:  # noqa: F841
```
Comment (webknjaz) [inline src/ansible_navigator/actions/config.py:324]: Wouldn't "fixing" the variable mean that `noqa` is no longer needed?

> **F841:** Local variable name is assigned to but never used

Comment (cidrblock) [reply src/ansible_navigator/actions/config.py:324]: yeah, brb

Comment (webknjaz) [reply src/ansible_navigator/actions/config.py:324]: Perhaps, the real fix is
```suggestion
                except KeyError:
```

**Review by webknjaz — APPROVED** [review]

**Pull request merged**